reposerver/repository/repository.go

```diff
@@ -285,6 +285,15 @@ func (s *Service) runRepoOperation(
 	} else {
 		gitClient, revision, err = s.newClientResolveRevision(repo, revision, git.WithCache(s.cache, !settings.noRevisionCache && !settings.noCache))
 		if err != nil {
+			if gitClient != nil && strings.Contains(err.Error(), "remote:") {
+				// https://github.com/go-git/go-git/issues/582
+				// get full error message from native git fetch command
+				if err := gitClient.Init(); err != nil {
+					log.Warnf("Failed to initialize git repo: %v", err)
+				} else if err := gitClient.Fetch(""); err != nil {
```

Comment:
At this point, do we necessarily have any reason to believe that a git fetch will return a relevant error message?